<launch>
  <!-- set simulation params -->
  <arg name="simulation_factor" default="1"/>
  <arg name="update_rate" default="25.0"/>
  <!-- select a scene -->
  <arg name="scene_file" default="$(find vehicle_simulator)scenes/corridor_env.xml"/>


  <!-- Simulator -->
  <node name="simulate_environment" pkg="vehicle_simulator" type="simulate_environment" output="screen">
    <param name="scene_file" value="$(arg scene_file)" type="string"/>
    <param name="simulation_factor" value="$(arg simulation_factor)" type="double"/>
    <param name="update_rate" value="$(arg update_rate)" type="double"/>
  </node>

  <!-- Fake map_sever -->
  <include file="$(find autosim_map)/launch/autosim_map.launch"/>
  <include file="$(find fake_localization)/launch/fake_localization.launch"/>

  <!-- Robot controller -->
  <include file="$(find vehicle_simulator)/launch/conflict.launch"/>

  <!-- Open interface_teleop -->
  <include file="$(find vehicle_simulator)/launch/interface_teleop.launch"/> 

  <!-- Description for the robot -->
  <include file="$(find wheeltec_description)/launch/robot_description.launch"/>

  <include file="$(find tocg)/launch/tocg_game.launch"/>

  <!-- Rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find vehicle_simulator)/rviz/corridor.rviz"/>

</launch>